### PR TITLE
Updating CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by drduhe**
*Issue #, if available:*

*Description of changes:*
Updating the contributing guide to be more exhaustive and include correct development workflows.   

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,59 +1,235 @@
...
 
+1. Make sure your `dev` branch is up-to-date
 
-## Finding contributions to work on
-Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any 'help wanted' issues is a great place to start.
+2. Create a feature branch based off of `dev`
 
+   ```
+   $ git checkout dev
+   $ git checkout -b feature-new-documentation
```
Comment (edparris) [inline CONTRIBUTING.md:56]: Update example to follow conventions described above: e.g. `git checkout -b feature/new-documentation`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,59 +1,235 @@
...
+   the hotfix PR.
+
+## Code Style
+
+We maintain a consistent code style throughout the project, so please ensure your changes align with our existing style.
+Take a look at the existing codebase to understand the patterns and conventions we follow.
+
+## Commit Messages
+
+Clear and concise commit messages are important for keeping the project history readable. Please follow these rules when
```
Comment (edparris) [inline CONTRIBUTING.md:219]: Update this to follow the Conventional Commits guidance. https://www.conventionalcommits.org/en/v1.0.0/#summary

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,59 +1,235 @@
...
+   $ git push
+   ```
+
+2. Finish off the tasks in the release checklist. Once everything is done, close
+   the hotfix PR.
+
+## Code Style
+
+We maintain a consistent code style throughout the project, so please ensure your changes align with our existing style.
+Take a look at the existing codebase to understand the patterns and conventions we follow.
```
Comment (edparris) [inline CONTRIBUTING.md:215]: Consider mentioning the automation provided by the `lint` tox environment.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,59 +1,235 @@
...
+   provides a safety net should your hard drive crash.
 
+    ```
+    $ ... make changes
+    $ git add -A .
+    $ git commit -m "Add new documentation files"
+    $ ... make more changes
+    $ git add -A .
+    $ git commit -m "Fix some spelling errors"
+    $ git push
```
Comment (edparris) [inline CONTRIBUTING.md:71]: I also generally squash multiple commits before putting out the final PR. It looks like GitHub has some automation about that built in so maybe it's not critical these days but something to think about.

File: CONTRIBUTING.md (hunk 2 above)
Comment (drduhe) [reply CONTRIBUTING.md:219]: Will add this.

File: CONTRIBUTING.md (hunk 3 above)
Comment (RanbirAulakh) [reply CONTRIBUTING.md:215]: +1 we should consider adding linting as a git prehooks requirement

File: CONTRIBUTING.md (hunk 1 above)
Comment (drduhe) [reply CONTRIBUTING.md:56]: Fixed

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,59 +1,235 @@
...
-3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+| Branch           | Protected? | Base Branch        | Description                                                                                                                                                                                                   |
+|:-----------------|:-----------|:-------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `main`           | YES        | N/A                | What is live in production (**stable**).<br/>A pull request is required to merge code into `main`.                                                                                                            |
+| `dev`            | YES        | `main`             | The latest state of development (**unstable**).                                                                                                                                                               |
```
[the remark this answers is not in the document]
Comment (drduhe) [reply CONTRIBUTING.md:39]: Fixed

**Review by RanbirAulakh — APPROVED** [review]

**Review by edparris — APPROVED** [review]

**Pull request merged**